src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/CalculateEarnings/CalculateEarningsCommandHandler.cs

```diff
@@ -26,6 +26,7 @@ public class CalculateEarningsCommandHandler : ICommandHandler<CalculateEarnings
         public async Task Handle(CalculateEarningsCommand command, CancellationToken cancellationToken = default)
         {
             var incentive = await _domainRepository.Find(command.ApprenticeshipIncentiveId);
+            if (incentive.Withdrawn) return;
```

Comment:
This "business logic" should go into the domain object .  To avoid calling the other two services unnecessarily change  the CalculateEarnings method to accept the IIncentivePaymentProfilesService  and ICollectionCalendarService parameters instead of the data.  Then it's the double dispatch pattern and all the business logic can be encapsulated in the domain object.

Reply:
agreed